include/deal2lkit/utilities.h

```diff
@@ -460,4 +461,26 @@ void smart_delete (SmartPointer<TYPE> &sp)
     }
 }
 
+/**
+ *  A simple class to shift a vector by a scalar.
+ *  This function is deprecated in deal but needed in many codes
+ */
+
+template <typename VEC>
+void vector_shift(VEC &in_vec, double a_scalar)
+{
+#ifdef DEAL_II_WITH_CXX11
+  for (auto i : in_vec.locally_owned_elements())
+    {
+      in_vec[i] += a_scalar;
+    }
+#else
+  for (IndexSet::ElementIterator it=in_vec.locally_owned_elements().begin(); it != in_vec.locally_owned_elements().end(); ++it)
```

Comment:
The operator locally_owned_elements returns a copy. It is not The same
objet on the two comparisons. You should call it outside:

IndexSet loc = vec.locally_owned...

For it = loc.begin(); it!=loc.end(); ++it

Luca

On 18/set/2015, at 17:46, ngiuliani <notifications@github.com> wrote:

In include/deal2lkit/utilities.h
<https://github.com/mathLab/deal2lkit/pull/133#discussion_r39869600>:

> @@ -460,4 +461,26 @@ void smart_delete (SmartPointer<TYPE> &sp)
>      }
>  }
>
> +/**
> + *  A simple class to shift a vector by a scalar.
> + *  This function is deprecated in deal but needed in many codes
> + */
> +
> +template <typename VEC>
> +void vector_shift(VEC &in_vec, double a_scalar)
> +{
> +#ifdef DEAL_II_WITH_CXX11
> +  for (auto i : in_vec.locally_owned_elements())
> +    {
> +      in_vec[i] += a_scalar;
> +    }
> +#else
> +  for (IndexSet::ElementIterator it=in_vec.locally_owned_elements().begin(); it != in_vec.locally_owned_elements().end(); ++it)

It complains here. It asserts that the two iterators compared in != are
from different sets...It seems strange to me...

—
Reply to this email directly or view it on GitHub
<https://github.com/mathLab/deal2lkit/pull/133/files#r39869600>.